src/tracer/MOM_offline_aux.F90

```diff
@@ -101,66 +101,63 @@ subroutine update_h_vertical_flux(G, GV, ea, eb, h_pre, h_new)
     enddo ; enddo
-
   enddo
 
 end subroutine update_h_vertical_flux
 
 !> This routine limits the mass fluxes so that the a layer cannot be completely depleted.
 !! NOTE: Only used in non-ALE mode
 subroutine limit_mass_flux_3d(G, GV, uh, vh, ea, eb, h_pre)
-  type(ocean_grid_type),   pointer       :: G     !< ocean grid structure
-  type(verticalGrid_type), pointer       :: GV    !< ocean vertical grid structure
+  type(ocean_grid_type),   intent(in)    :: G     !< ocean grid structure
+  type(verticalGrid_type), intent(in)    :: GV    !< ocean vertical grid structure
   real, dimension(SZIB_(G),SZJ_(G),SZK_(GV)), &
-                           intent(inout) :: uh    !< Mass flux through zonal face [kg]
+                           intent(inout) :: uh    !< Mass flux through zonal face [H L2 ~> m3 or kg]
   real, dimension(SZI_(G),SZJB_(G),SZK_(GV)), &
-                           intent(inout) :: vh    !< Mass flux through meridional face [kg]
+                           intent(inout) :: vh    !< Mass flux through meridional face [H L2 ~> m3 or kg]
   real, dimension(SZI_(G),SZJ_(G),SZK_(GV)), &
                            intent(inout) :: ea    !< Mass of fluid entrained from the layer
-                                                  !! above within this timestep [kg m-2]
+                                                  !! above within this timestep [H ~> m or kg m-2]
   real, dimension(SZI_(G),SZJ_(G),SZK_(GV)), &
                            intent(inout) :: eb    !< Mass of fluid entrained from the layer
-                                                  !! below within this timestep [kg m-2]
+                                                  !! below within this timestep [H ~> m or kg m-2]
   real, dimension(SZI_(G),SZJ_(G),SZK_(GV)), &
                            intent(in)    :: h_pre !< Layer thicknesses at the end of the previous
-                                                  !! step [kg m-2].
+                                                  !! step [H ~> m or kg m-2]
 
   ! Local variables
+  real, dimension(SZI_(G),SZJ_(G),SZK_(GV)) :: top_flux    ! Net upward fluxes through the layer
+                                                           ! top [H ~> m or kg m-2]
+  real, dimension(SZI_(G),SZJ_(G),SZK_(GV)) :: bottom_flux ! Net downward fluxes through the layer
+                                                           ! bottom [H ~> m or kg m-2]
+  real :: pos_flux ! Net flux out of cell [H L2 ~> m3 or kg]
+  real :: hvol     ! Cell volume [H L2 ~> m3 or kg]
+  real :: scale_factor  ! A nondimensional rescaling factor between 0 and 1 [nondim]
+  real :: max_off_cfl   ! The maximum permitted fraction that can leave in a timestep [nondim]
   integer :: i, j, k, m, is, ie, js, je, nz
-  real, dimension(SZI_(G),SZJ_(G),SZK_(GV)) :: top_flux, bottom_flux
-  real :: pos_flux, hvol, h_neglect, scale_factor, max_off_cfl
 
-  max_off_cfl =0.5
+  max_off_cfl = 0.5
```

Comment:
This probably should be made a runtime parameter

Reply:
Yes, it should be a runtime parameter, but for now let's leave this until you get a chance to create the separate driver and can test the new option.

The other possibility is that we can take care of this as a part of a broader sweep through the MOM6 code to replace all of the significant hard-coded parameters with runtime parameters.  However, that step will require some rethinking about how (or whether) we do version control in MOM6-examples on the various MOM_parameter_doc.all files.  Look for a discussion of this in the new year.